src/iframe-helper.js

```diff
@@ -136,6 +137,9 @@ function getListenForEvents(parentWin, sentinel, origin, triggerWin) {
       // 3P code path, we may accept messages from nested frames.
       windowEvents = we;
       break;
+    } else {
+      windowEvents = we;
+      break;
```

Comment:
scratch that, the last commit broke a lot of tests, reverted and slowly adding back in changes. will ping you on hangouts